internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -799,12 +800,13 @@ func (c *Coordinator) DiagnosticHooks() diagnostics.Hooks {
 					}
 				}
 				output := StateHookOutput{
-					State:        s.State,
-					Message:      s.Message,
-					FleetState:   s.FleetState,
-					FleetMessage: s.FleetMessage,
-					LogLevel:     s.LogLevel,
-					Components:   compStates,
+					State:          s.State,
+					Message:        s.Message,
+					FleetState:     s.FleetState,
+					FleetMessage:   s.FleetMessage,
+					LogLevel:       s.LogLevel,
+					Components:     compStates,
+					UpgradeDetails: s.UpgradeDetails,
```

Comment:
When is this field present? Do we get details about the upgrade after a failure ? Or those get cleared as soon as the grace period is over ?

Reply:
> When is this field present? 

This field is populated when the upgrade process starts: https://github.com/elastic/elastic-agent/blob/2a93086d5c7695f9a08e118667dac0da677157c6/internal/pkg/agent/application/coordinator/coordinator.go#L475-L476

> Do we get details about the upgrade after a failure ?

Yes, if the upgrade process fails at any point, this field will continue to remain populated. 

> Or those get cleared as soon as the grace period is over ?

I assume you are referring to the Upgrade Watcher's grace period.

Exactly when we clear it out after that is not yet implemented. We need to decide whether we want to clear it out based on some timeout (maybe the grace period or maybe something even beyond that) or due to some user intervention (which could be retrying the upgrade or something else just to clear the upgrade details, not sure yet) or either (whichever happens first).